src/node/element/path/data.rs

```diff
@@ -185,52 +185,71 @@ impl<'l> Parser<'l> {
+            'l' => Line(Relative, self.read_parameters(false)?.into()),
 
-            'H' => HorizontalLine(Absolute, parameters),
-            'h' => HorizontalLine(Relative, parameters),
+            'H' => HorizontalLine(Absolute, self.read_parameters(false)?.into()),
+            'h' => HorizontalLine(Relative, self.read_parameters(false)?.into()),
 
-            'V' => VerticalLine(Absolute, parameters),
-            'v' => VerticalLine(Relative, parameters),
+            'V' => VerticalLine(Absolute, self.read_parameters(false)?.into()),
+            'v' => VerticalLine(Relative, self.read_parameters(false)?.into()),
 
-            'Q' => QuadraticCurve(Absolute, parameters),
-            'q' => QuadraticCurve(Relative, parameters),
+            'Q' => QuadraticCurve(Absolute, self.read_parameters(false)?.into()),
+            'q' => QuadraticCurve(Relative, self.read_parameters(false)?.into()),
 
-            'T' => SmoothQuadraticCurve(Absolute, parameters),
-            't' => SmoothQuadraticCurve(Relative, parameters),
+            'T' => SmoothQuadraticCurve(Absolute, self.read_parameters(false)?.into()),
+            't' => SmoothQuadraticCurve(Relative, self.read_parameters(false)?.into()),
 
-            'C' => CubicCurve(Absolute, parameters),
-            'c' => CubicCurve(Relative, parameters),
+            'C' => CubicCurve(Absolute, self.read_parameters(false)?.into()),
+            'c' => CubicCurve(Relative, self.read_parameters(false)?.into()),
 
-            'S' => SmoothCubicCurve(Absolute, parameters),
-            's' => SmoothCubicCurve(Relative, parameters),
+            'S' => SmoothCubicCurve(Absolute, self.read_parameters(false)?.into()),
+            's' => SmoothCubicCurve(Relative, self.read_parameters(false)?.into()),
 
-            'A' => EllipticalArc(Absolute, parameters),
-            'a' => EllipticalArc(Relative, parameters),
+            'A' => EllipticalArc(Absolute, self.read_parameters(true)?.into()),
+            'a' => EllipticalArc(Relative, self.read_parameters(true)?.into()),
 
             'Z' | 'z' => Close,
 
             _ => raise!(self, "found an unknown path command '{}'", name),
         }))
     }
 
-    fn read_parameters(&mut self) -> Result<Vec<Number>> {
+    fn read_parameters(&mut self, is_arc: bool) -> Result<Vec<Number>> {
         let mut parameters = Vec::new();
+        let mut cmd_index: usize = 0;
 
-        while let Some(number) = self.read_number()? {
+        while let Some(number) = self.read_number_or_flag(if is_arc { Some(cmd_index % 7) } else { None })? {
+            cmd_index = cmd_index + 1;
             parameters.push(number);
             self.reader.consume_whitespace();
             self.reader.consume_char(',');
         }
+
         Ok(parameters)
     }
 
+    fn read_number_or_flag(&mut self, arc_relative_index: Option<usize>) -> Result<Option<Number>> {
+        match arc_relative_index {
+            Some(x) if x == 3 || x == 4 => self.read_flag(),
```

Comment:
Could you please explain a bit why it is modulo 7 there and then 3 and 4 here?

Reply:
Sure. A single set of parameters for Elliptical Arc command has [7 elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d). The flag parameters are always in positions 3 and 4.

An Elliptical Arc command may also have any arbitrary count of parameter sets. Each set must have 7 elements. The flag parameters will appear in positions 3 and 4 within each set.

The top level function (now `read_parameters_elliptical_arc`) parses through all of the parameter sets, for example, in this test case, which has two sets of parameters for the one `a` command:
`48 48 0 1148-48 48 48 0 01-48 48`

It needs to use flag parsing logic on the 3rd and 4th parameters within each set of 7, and regular number parsing logic for all other parameters.

Reply:
After catching an off-by-one error in my explanation, and then looking again at the code, I thought it might be clearer to just make this a 1-based index and go with `4` and `5` in the code instead of `3` and `4`. So I just pushed another commit for that.

Reply:
And...I think it's still wrong. I want to add more test cases with additional sets of parameters.

Reply:
Ok, maybe not wrong. I just added another test case with three sets of params which is passing.

Reply:
Thanks!